hpx/plugins/parcelport/tcp/receiver.hpp

```diff
@@ -107,7 +107,7 @@ namespace hpx { namespace parcelset { namespace policies { namespace tcp
                     = &receiver::handle_read_header<Handler>;
 
                 boost::asio::async_read(socket_, buffers,
-                    boost::bind(f, shared_from_this(),
+                    util::bind(f, shared_from_this(),
                         boost::asio::placeholders::error,
                         boost::asio::placeholders::bytes_transferred,
                         util::make_tuple(handler)));
```

Comment:
Removing the workaround leads to compile time errors when ASIO checks for handler requirements. It appears to be a bug in ASIO's concept check code that instantiates more stuff than it should.